### ISSUE TITLE
Support non W3C-standards capabilities in Test runner configurations

### ISSUE THREAD

**Issue by osddeitf** [webdriverio/webdriverio#6572]
## Is your feature request related to a problem? Please describe
We want to setup our projects with e2e tests, but want to avoid lengthy setups and code as much as possible.

Then, we found this awesome project: https://github.com/webdriverio/cucumber-boilerplate.
Its README said that we need **Selenium Grid** to work, but our current infrastructures using self-host **Browserless**.
Luckily, I found this: https://docs.browserless.io/docs/selenium-library.html.
Seems like **Selenium** could connect to **Browserless**.
And I learned that there is `web-driver protocol`, that webdriver.io support.

So that in `cucumber-boilerplate` project, I edited `wdio.config.js` like this:
```js
{
    runner: 'local',
    protocol: 'http',
    hostname: 'localhost',
    port: 3000,
    path: '/webdriver',
    capabilities: [{
        maxInstances: 5,
        browserName: 'chrome',
        // Token to connect to Browserless instance
        'browserless.token': '1bxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
    }],
}
```
If `browserless.token` in the above config commented out, and my **Browserless** instance configured without token, it did work.
But when I try to set `browserless.token` in the capabilities, it didn't work.

I wonder why the example in [Browserless doc](https://docs.browserless.io/docs/selenium-library.html) work, and try to copy and run myself, it worked.
After inspecting logs of Browserless instance, I found that if I connect using `selenium-webdriver`, the capabilities sent to server are:
```
[1615280732.884][INFO]: [160e34562d75eb9652aa0f32587b85c3] COMMAND InitSession {
"capabilities": {
  "alwaysMatch": {
    "browserName": "chrome",
    "goog:chromeOptions": {
      "args": [ "--disable-background-timer-throttling", "--disable-backgrounding-occluded-windows", "--disable-breakpad", "--disable-component-extensions-with-background-pages", "--disable-dev-shm-usage", "--disable-extensions", "--disable-features=TranslateUI,BlinkGenPropertyTrees", "--disable-ipc-flooding-protection", "--disable-renderer-backgrounding", "--enable-features=NetworkService,NetworkServiceInProcess", "--force-color-profile=srgb", "--hide-scrollbars", "--metrics-recording-only", "--mute-audio", "--headless", "--no-sandbox", "--user-data-dir=/tmp/browserless-data-dir-f9AJJS" ],
      "binary": "/usr/bin/google-chrome"
    }
  }
},
"desiredCapabilities": {
  "browserName": "chrome",
  "browserless.token": "9f064794-b615-4a6c-be5b-47299ff0666e",
  "goog:chromeOptions": {
    "args": [ "--disable-background-timer-throttling", "--disable-backgrounding-occluded-windows", "--disable-breakpad", "--disable-component-extensions-with-background-pages", "--disable-dev-shm-usage", "--disable-extensions", "--disable-features=TranslateUI,BlinkGenPropertyTrees", "--disable-ipc-flooding-protection", "--disable-renderer-backgrounding", "--enable-features=NetworkService,NetworkServiceInProcess", "--force-color-profile=srgb", "--hide-scrollbars", "--metrics-recording-only", "--mute-audio", "--headless", "--no-sandbox", "--user-data-dir=/tmp/browserless-data-dir-f9AJJS" ],
    "binary": "/usr/bin/google-chrome"
  }
}
}
```

But, when I'm using **Webdriver.io** with the above `wdio.config.js`, it's like that:
```
[1615280761.025][INFO]: [7994d42ed597904defb1498f4c3b4a18] COMMAND InitSession {
"capabilities": {
    "alwaysMatch": {
      "browserName": "chrome",
      "browserless.token": "1bxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
    },
    "firstMatch": [ {

    } ]
  },
  "desiredCapabilities": {
    "browserName": "chrome",
    "browserless.token": "1bxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
  }
}
[1615280762.042][INFO]: [66121a7c1454f9ac5684dbee84609997] RESPONSE InitSession ERROR invalid argument: unrecognized capability: browserless.token
```
And there is an error:
```
[1615280762.042][INFO]: [66121a7c1454f9ac5684dbee84609997] RESPONSE InitSession ERROR invalid argument: unrecognized capability: browserless.token
```

I noticed there is a different that using `selenium-webdriver`, `capabilities.alwaysMatch["browserless.token"]` is not set.
Eventually, I decided to read the underlying code of `selenium-webdriver` to see what makes the different.
I found in `webdriver.js` that:
```js
function filterNonW3CCaps(capabilities) {
  let newCaps = new Capabilities(capabilities)
  for (let k of newCaps.keys()) {
    // Any key containing a colon is a vendor-prefixed capability.
    if (!(W3C_CAPABILITY_NAMES.has(k) || k.indexOf(':') >= 0)) {
      newCaps.delete(k)
    }
  }
  return newCaps
}

static createSession(executor, capabilities, onQuit = undefined) {
    let cmd = new command.Command(command.Name.NEW_SESSION)

    // For OSS remote ends.
    cmd.setParameter('desiredCapabilities', capabilities)
    // For W3C remote ends.
    cmd.setParameter('capabilities', {
      alwaysMatch: filterNonW3CCaps(capabilities),
    })

    let session = executeCommand(executor, cmd)
    if (typeof onQuit === 'function') {
      session = session.catch((err) => {
        return Promise.resolve(onQuit.call(void 0)).then((_) => {
          throw err
        })
      })
    }
    return new this(session, executor, onQuit)
  }
```
They use `filterNonW3CCaps` to filter out the *NonW3CCaps*, which in my case is `browserless.token`, before sending out the capabilities.

I tried to find various resources, and found that [Webdriver.io testrunner configuration](https://webdriver.io/docs/configurationfile) do not have a separate `desiredCapabilities` config, so I'm hopeless.

## Describe the solution you'd like
I'd like there are some mechanisms to opt-out the non-standards like `selenium-webdriver` package did.
This should open the user more choice over the providers of cloud service like **Browserless**.

**Comment by christian-bromann** [comment]
I raised an issue in the [browserless docs](https://github.com/browserless/docs/issues/1). Ultimately this is a bug in the browserless platform using an invalid W3C capability. The JSONWP has been deprecated for a long time and so are random capability definitions. A lot of platforms allow to still mix them which doesn't help the ecosystem as it creates tremendous confusions and unexpected behaviors.

I am split on this. While I would like to offer the flexibility to use whatever protocol you want (as we did for WebDriver and Chrome DevTools) I am unsure about re-allowing use of a deprecated protocol, let alone continuing to allow a mixture of both.

I can recommend using [WebdriverIO@v4](http://v4.webdriver.io/) where we still use the JSONWireProtocol.

Closing this as I don't see a good way to allow this in WebdriverIO. Happy to have a [discussion](https://github.com/webdriverio/webdriverio/discussions/new) about the topic mentioned above. Thanks for raising the issue.
